examples/online-application-portal/response_err_interceptor.bal

```diff
@@ -0,0 +1,77 @@
+// Copyright (c) 2024 WSO2 LLC. (http://www.wso2.com).
+//
+// WSO2 LLC. licenses this file to you under the Apache License,
+// Version 2.0 (the "License"); you may not use this file except
+// in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+import ballerina/constraint;
+import ballerina/http;
+
+service class ResponseErrorInterceptor {
+    *http:ResponseErrorInterceptor;
+
+    remote function interceptResponseError(http:RequestContext ctx, error err) returns ClientError|ServerError|error {
+        string path = check ctx.getWithType("rawPath");
+        string method = check ctx.getWithType("method");
+
+        if err is constraint:Error {
+            return <ClientError>{
+                body: {
+                    code: http:STATUS_BAD_REQUEST,
+                    message: err.message(),
+                    path: path,
+                    method: method
+                }
+            };
+        }
+
+        if err is http:ClientRequestError {
+            http:Detail responseDetails = err.detail();
+            if responseDetails.statusCode == http:STATUS_BAD_REQUEST {
```

Comment:
Are you only going to handle the bad request here? For all other client request errors you are returning a internal server error in L#58. Is that your requirement?

Reply:
Yes. In the backend only returns two types of errors HTTP 400 or HTTP 500. Since, I have already checked for those in the previous block all other errors would be considered as server errors.